cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/distributed/AbstractTwillProgramController.java

```diff
@@ -119,7 +119,7 @@ protected final void doStop() throws Exception {
     Futures.getUnchecked(twillController.terminate());
   }
 
-  protected final TwillController getTwillController() {
+  public final TwillController getTwillController() {
```

Comment:
You shouldn't expose this directly. You could have method on this class to help changing log level. You can introduce a new interface, something like `LogLevelUpdater` that has the update and reset methods and have this class implements it (the implementation can us the twill controller). Then from the caller side, you don't do instanceof against `AbstractTwillProgramController`, but instead against `LogLevelUpdater` which is much clearer on what the intention is.